src/eg/src/verifiable_decryption.rs

```diff
@@ -0,0 +1,682 @@
+#![deny(clippy::unwrap_used)]
+#![deny(clippy::expect_used)]
+#![deny(clippy::panic)]
+#![deny(clippy::manual_assert)]
+
+use itertools::izip;
+use num_bigint::BigUint;
+use num_traits::{One, Zero};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use util::{
+    csprng::Csprng,
+    integer_util::{
+        get_single_coefficient, group_lagrange_at_zero, mod_inverse, to_be_bytes_left_pad,
+    },
+};
+
+use crate::{
+    discrete_log::DiscreteLog,
+    election_parameters::ElectionParameters,
+    fixed_parameters::FixedParameters,
+    guardian::GuardianIndex,
+    guardian_public_key::GuardianPublicKey,
+    guardian_share::GuardianSecretKeyShare,
+    hash::{eg_h, HValue},
+    joint_election_public_key::{Ciphertext, JointElectionPublicKey},
+};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct DecryptionShare {
```

Comment:
Maybe add a bit of documentation here

Reply:
I have added overall more documentation.